packages/l/lua/xmake.lua

```diff
@@ -26,10 +26,31 @@ package("lua")
 
     on_install(function (package)
         local sourcedir = os.isdir("src") and "src/" or "" -- for tar.gz or git source
+        if is_plat("iphoneos", "android") then
+            -- disable system() calls as they're not supported on these platforms
+            io.replace(sourcedir .. "/luaconf.h",
+[[/* =================================================================== */
+
+/*
+** Local configuration. You can use this space to add your redefinitions]], 
+[[#if defined(LUA_CORE)
+/* disable system calls as it's not available on this sytem */
+#define system(s) ((s)==NULL ? 0 : -1)
+#endif
+
+/* =================================================================== */
+
+/*
+** Local configuration. You can use this space to add your redefinitions
+]], {plain = true})
```

Comment:
too long patch, you can use add_patches or io.open("", "a")/file:write to append content to tail

Reply:
I'm not sure I can write a patch that will work with all versions of Lua, but I'll try.

Also this can't be tail since it has to be before the last #endif